### ISSUE TITLE
Unable to type within container terminal in PHPStorm

### ISSUE THREAD

**Issue by tehmuh** [docker/for-win#12818]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 1B5F6448-3B09-46A6-A06B-6B2108803A35/20220705141122

### Actual behavior
Not able to type anything within a container terminal in PHPStorm. Docker Desktop is connected to PHPStorm with the Docker integration plugin. Connecting to a container works fine though.
Connecting manually in the "Terminal"-tab with "docker exec -it xxx bash" works fine.

### Expected behavior
I expect to be able to write inside the console.

### Information
  - Is it reproducible? Yes
  - Is the problem new? Yes, as far as I know. I did not find a bug report describing this issue.
  - Did the problem appear with an update? This problem occurs since 4.10.0 (82025). A downgrade to 4.9.1 "solves" this problem.
  - Windows Version:  Windows 10 Pro 21H1 19043.1110
  - Docker Desktop Version:  4.10.0
  - WSL2 or Hyper-V backend? WSL2, Ubuntu 20.04
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: No

### Output of `& "C:\Program Files\Docker\Docker\resources\com.docker.diagnose.exe" check`
[2022-07-05T14:22:14.311623800Z][com.docker.diagnose.exe][I] set path configuration to OnHost
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0031: does the Docker API work?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0001: is the application running?
[SKIP] DD0018: does the host support virtualization?
[PASS] DD0002: does the bootloader have virtualization enabled?
[PASS] DD0017: can a VM be started?
[FAIL] DD0024: is WSL installed? exit status 0xffffffff
[PASS] DD0021: is the WSL 2 Windows Feature enabled?
[PASS] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[PASS] DD0025: are WSL distros installed?
[PASS] DD0026: is the WSL LxssManager service running?
[PASS] DD0029: is the WSL 2 Linux filesystem corrupt?
[PASS] DD0035: is the VM time synchronized?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0013: is the $PATH ok?
[FAIL] DD0005: is the user in the docker-users group? Die angegebene Domäne ist nicht vorhanden, oder es konn
[PASS] DD0007: is the backend responding?
[FAIL] DD0014: are the backend processes running? 1 error occurred:
        * vpnkit.exe is not running

[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[PASS] DD0006: is the Docker Desktop Service responding?
[PASS] DD0012: is the VM networking working?
[PASS] DD0032: do Docker networks overlap with host IPs?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0033: does the host have Internet access?

Please investigate the following 1 issue:

1 : The test: is WSL installed?
    Failed with: exit status 0xffffffff`
### Steps to reproduce the behavior
  1. Start Docker Desktop
  2. Start container
  3. Start PHPStorm
  4. Connect to Docker Desktop within "Services"
  5. Create terminal of a container

**Comment by Carobit** [comment]
I have the same issue. After upgrading docker desktop to 4.10.1 the docker ternimal is not responding in PhpStorm.  The terminal opens and cannot type/interact anymore.
Log from docker:
```
[2022-07-14T10:28:56.420777100Z][com.docker.proxy.exe][I] (5368a181) 2528b98e-ProxyCMD C->S BackendAPI POST /usage: {"command":"containerExecOthersLinux","count":1}
[2022-07-14T10:28:56.421859800Z][com.docker.backend.exe][I] (5368a181) b14d15d7-BackendAPI S<-C 2528b98e-ProxyCMD POST /usage
[2022-07-14T10:28:56.422365900Z][GoBackendProcess  ][Info   ] [common/pkg/backend.(*backendClient).UpdateUsage(0xc00007a0f0, {{0xc0000364b0, 0x18}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, 0x1})
[2022-07-14T10:28:56.422365900Z][GoBackendProcess  ][Info   ] [	common/pkg/backend/client.go:475 +0xc2
[2022-07-14T10:28:56.422365900Z][GoBackendProcess  ][Info   ] [common/pkg/proxy.(*BackendAnalytics).sendCommand(0xc0001e4800, {0x152af9e, 0xd}, {0x1523dac, 0x6}, 0x1)
[2022-07-14T10:28:56.422365900Z][GoBackendProcess  ][Info   ] [	common/pkg/proxy/analytics.go:104 +0x10e
[2022-07-14T10:28:56.422365900Z][GoBackendProcess  ][Info   ] [common/pkg/proxy.(*BackendAnalytics).trackRouteForHTTPMethod(0x14ef2a0?, 0xc0000c5200, 0xc000e14e10?, 0x157bed0)
[2022-07-14T10:28:56.422365900Z][GoBackendProcess  ][Info   ] [	common/pkg/proxy/analytics.go:97 +0xf4
[2022-07-14T10:28:56.422365900Z][GoBackendProcess  ][Info   ] [common/pkg/proxy.(*BackendAnalytics).Middleware.func1({0x1654a10, 0xc0004ea460}, 0xc0000c5200)
[2022-07-14T10:28:56.422365900Z][GoBackendProcess  ][Info   ] [	common/pkg/proxy/analytics.go:59 +0x8d
[2022-07-14T10:28:56.422365900Z][GoBackendProcess  ][Info   ] [github.com/gorilla/mux.(*Router).ServeHTTP(0xc0002b00c0, {0x1654a10, 0xc0004ea460}, 0xc000e12300)
[2022-07-14T10:28:56.422365900Z][GoBackendProcess  ][Info   ] [	vendor/github.com/gorilla/mux/mux.go:210 +0x1cf
[2022-07-14T10:28:56.422894500Z][com.docker.proxy.exe][I] (5368a181) 2528b98e-ProxyCMD C<-S b14d15d7-BackendAPI POST /usage (2.1174ms): OK
[2022-07-14T10:28:56.423424000Z][com.docker.proxy.exe][I] usage containerExecOthersLinux + 1
[2022-07-14T10:28:56.423424000Z][com.docker.proxy.exe][I] proxy >> POST /v1.40/containers/6b9c73d0208a51e705e59dd95ea4c7cbedb4fa10d8ef25b9b76300c4758f38ec/exec
[2022-07-14T10:28:56.425562800Z][com.docker.proxy.exe][I] proxy << POST /v1.40/containers/6b9c73d0208a51e705e59dd95ea4c7cbedb4fa10d8ef25b9b76300c4758f38ec/exec (2.1388ms)
[2022-07-14T10:28:56.428188400Z][com.docker.proxy.exe][I] (1eb11bcf) 2528b98e-ProxyCMD C->S BackendAPI POST /usage: {"command":"containerStartOthersLinux","count":1}
[2022-07-14T10:28:56.429858700Z][com.docker.proxy.exe][I] (1eb11bcf) 2528b98e-ProxyCMD C<-S b14d15d7-BackendAPI POST /usage (1.6703ms): OK
[2022-07-14T10:28:56.430385300Z][com.docker.proxy.exe][I] usage containerStartOthersLinux + 1
[2022-07-14T10:28:56.430385300Z][com.docker.proxy.exe][I] proxy >> POST /v1.40/exec/51d433931e68e6083bd7082c2aba9ed1a5e5ab5454a1ed75a80fcba68823cd53/start
[2022-07-14T10:28:56.446363100Z][com.docker.proxy.exe][I] proxy >> POST /v1.40/exec/51d433931e68e6083bd7082c2aba9ed1a5e5ab5454a1ed75a80fcba68823cd53/resize?h=7&w=158
[2022-07-14T10:28:56.513251400Z][com.docker.proxy.exe][I] proxy << POST /v1.40/exec/51d433931e68e6083bd7082c2aba9ed1a5e5ab5454a1ed75a80fcba68823cd53/resize?h=7&w=158 (66.8883ms)
[2022-07-14T10:28:56.422365900Z][com.docker.backend.exe][I] (5368a181) b14d15d7-BackendAPI S<-C 2528b98e-ProxyCMD bind: {"command":"containerExecOthersLinux","count":1}
[2022-07-14T10:28:56.422894500Z][com.docker.backend.exe][I] (5368a181) b14d15d7-BackendAPI S->C 2528b98e-ProxyCMD POST /usage (1.0347ms): OK
[2022-07-14T10:28:56.429318500Z][com.docker.backend.exe][I] (1eb11bcf) b14d15d7-BackendAPI S<-C 2528b98e-ProxyCMD POST /usage
[2022-07-14T10:28:58.426528300Z][GoBackendProcess  ][Info   ] [common/pkg/backend.(*backendClient).UpdateUsage(0xc00007a0f0, {{0xc000898240, 0x19}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, 0x1})
[2022-07-14T10:28:58.426528300Z][GoBackendProcess  ][Info   ] [	common/pkg/backend/client.go:475 +0xc2
[2022-07-14T10:28:58.426528300Z][GoBackendProcess  ][Info   ] [common/pkg/proxy.(*BackendAnalytics).sendCommand(0xc0001e4800, {0x152bd0f, 0xe}, {0x1523dac, 0x6}, 0x1)
[2022-07-14T10:28:58.426528300Z][GoBackendProcess  ][Info   ] [	common/pkg/proxy/analytics.go:104 +0x10e
[2022-07-14T10:28:58.426528300Z][GoBackendProcess  ][Info   ] [common/pkg/proxy.(*BackendAnalytics).trackRouteForHTTPMethod(0x1655188?, 0xc0000c4400, 0x13e8720?, 0x157bed0)
[2022-07-14T10:28:58.426528300Z][GoBackendProcess  ][Info   ] [	common/pkg/proxy/analytics.go:97 +0xf4
[2022-07-14T10:28:58.426528300Z][GoBackendProcess  ][Info   ] [common/pkg/proxy.(*BackendAnalytics).Middleware.func1({0x1654a10, 0xc0004ea1c0}, 0xc0000c4400)
[2022-07-14T10:28:58.426528300Z][GoBackendProcess  ][Info   ] [	common/pkg/proxy/analytics.go:59 +0x8d
[2022-07-14T10:28:58.426528300Z][GoBackendProcess  ][Info   ] [github.com/gorilla/mux.(*Router).ServeHTTP(0xc0002b00c0, {0x1654a10, 0xc0004ea1c0}, 0xc0000c4000)
[2022-07-14T10:28:58.426528300Z][GoBackendProcess  ][Info   ] [	vendor/github.com/gorilla/mux/mux.go:210 +0x1cf
[2022-07-14T10:28:56.429318500Z][com.docker.backend.exe][I] (1eb11bcf) b14d15d7-BackendAPI S<-C 2528b98e-ProxyCMD bind: {"command":"containerStartOthersLinux","count":1}
[2022-07-14T10:28:56.429858700Z][com.docker.backend.exe][I] (1eb11bcf) b14d15d7-BackendAPI S->C 2528b98e-ProxyCMD POST /usage (540.2µs): OK
```

log from PhpStorm:
```
2022-07-14 12:35:26,097 [  56708]   INFO - c.j.t.TerminalStarter - Terminal exiting
2022-07-14 12:35:26,098 [  56709] SEVERE - c.g.d.a.a.ResultCallbackTemplate - Error during callback
java.io.EOFException
	at okio.RealBufferedSource.require(RealBufferedSource.java:65)
	at okio.RealBufferedSource.readHexadecimalUnsignedLong(RealBufferedSource.java:307)
	at okhttp3.internal.http1.Http1ExchangeCodec$ChunkedSource.readChunkSize(Http1ExchangeCodec.java:492)
	at okhttp3.internal.http1.Http1ExchangeCodec$ChunkedSource.read(Http1ExchangeCodec.java:471)
	at okhttp3.internal.connection.Exchange$ResponseBodySource.read(Exchange.java:286)
	at okio.RealBufferedSource$1.read(RealBufferedSource.java:447)
	at java.base/java.io.InputStream.read(InputStream.java:205)
	at com.github.dockerjava.core.FramedInputStreamConsumer.accept(FramedInputStreamConsumer.java:41)
	at com.github.dockerjava.core.FramedInputStreamConsumer.accept(FramedInputStreamConsumer.java:12)
	at com.github.dockerjava.core.DefaultInvocationBuilder.lambda$executeAndStream$1(DefaultInvocationBuilder.java:275)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

Hope this helps.

**Comment by nicks** [comment]
If it works OK from the CLI but not in PHPStorm, that points to a bug in PHPStorm's Docker client...is it open source?

The log that @Carobit posted is interesting, particularly that it's using the v1.40 engine API rather than the v1.41 engine API.

I wonder if the problem is that PHPStorm is treating "Docker 4.10" as "Docker 4.1", similar to the old "Windows 9 vs Windows 95" problem https://www.pcworld.com/article/435584/why-windows-10-isnt-named-9-windows-95-legacy-code.html

**Comment by nicks** [comment]
Tracked against PHPStorm here: https://youtrack.jetbrains.com/issue/IDEA-297446/Unable-to-type-in-Services-Docker-container-Terminal-window-with-Docker-4101

**Comment by Carobit** [comment]
The bug seems to be fixed now at Jetbrains, in the mentioned issue. https://youtrack.jetbrains.com/issue/IDEA-297446/Unable-to-type-in-Services-Docker-container-Terminal-window-with-Docker-4101#focus=Comments-27-6310507.0-0
Waiting for a release to test it.

**Comment by nicks** [comment]
The comments on the jetbrains issue confirm this is fixed in PHPStorm 2022.2, closing
